### PR TITLE
feat(auth): add Two-Factor Authentication support

### PR DESCRIPTION
**Pull request by s0up4200**
Key Features:
- TOTP-based two-factor authentication implementation
- QR code generation for easy setup with authenticator apps
- Automatic cleanup of incomplete 2FA setups
- NTP time synchronization for reliable TOTP validation

Technical Details:
- Added 2FA-related db migrations
- Implemented backend services for 2FA secret generation and verification
- Created new API endpoints for 2FA setup
- Added frontend support for 2FA setup and verification flow
- Added NTP time sync verification during 2FA setup
- Added NTP-adjusted time fallback for login validation
- Added tests to `internal/auth/service_test.go`, `database/user_test.go` and `http/auth_test.go`

<details>
<summary>Click to see UI</summary>

CleanShot 2024-10-24 at 21 09 58@2x

CleanShot 2024-10-24 at 21 10 33@2x

CleanShot 2024-10-24 at 21 11 07@2x

</details>

### CODE REVIEW

File: internal/auth/service.go [hunk 1]
```diff
@@ -22,20 +28,48 @@ type Service interface {
...
+	Disable2FA(ctx context.Context, username string) error
 }
 
 type service struct {
 	log     zerolog.Logger
 	userSvc user.Service
 }
 
+var (
+	ntpServer = "pool.ntp.org"
```
[the remark this answers is not in the document]
Comment (s0up4200) [reply internal/auth/service.go:44]: yeah hitting that should be round robin

```go
package main

import (
	"fmt"
	"time"

	"github.com/beevik/ntp"
)

func main() {
	response, err := ntp.Query("pool.ntp.org")
	// or specific numbered pools:
	// response, err := ntp.Query("0.pool.ntp.org")
	// response, err := ntp.Query("1.pool.ntp.org")
	// response, err := ntp.Query("2.pool.ntp.org")
	// response, err := ntp.Query("3.pool.ntp.org")

	if err != nil {
		fmt.Printf("Error: %v\n", err)
		return
	}

	time := time.Now().Add(response.ClockOffset)
	fmt.Printf("Current time: %v\n", time)
}
```

```console
go run .
Current time: 2024-10-25 07:03:22.805100762 +0200 CEST m=+0.111432846
```

[the remark this answers is not in the document]
Comment (s0up4200) [reply internal/auth/service.go:44]: We should maybe hit `0.beevik-ntp.pool.ntp.org` instead, which is the maintainers' pool? @KyleSanderson

Comment (s0up4200) [inline internal/auth/service.go:44]:
```suggestion
	ntpServer = "0.beevik-ntp.pool.ntp.org"
```

Comment (s0up4200) [reply internal/auth/service.go:44]: Source: https://github.com/beevik/ntp?tab=readme-ov-file#querying-the-current-time

**Comment by martylukyy** [comment]
We already have a classnames helper importable by `import { classNames } from "@utils"`.
The added classnames package should be removed and replaced by the already existing one.

**Comment by s0up4200** [comment]
Closing in favor of #1853 which is a more proper solution for the project.

**Pull request closed without merging**